### ISSUE TITLE
jsons should be stricter on loading None

### ISSUE THREAD

**Issue by ramonhagenaars** [ramonhagenaars/jsons#60]
When deserializing, ``jsons`` accepts ``None`` when not in "strict-mode". It should not accept ``None`` at all, except when the target class is one of ``Any``, ``NoneType``, ``Optional[...]``, ``Union[None, ...]``.

Rationale:

```python
# Suppose some_obj happens to be None:
loaded_obj = jsons.load(some_obj, SomeClass)  # loaded_obj is now None.
...
# Somewhere else in your code, you do:
loaded_obj.some_method()  # This raises.
```
The above example shows how accepting a ``None`` may be confusing and possibly lead a developer in the wrong direction by falsely declaring the deserialization to be successful. If ``jsons.load(some_obj, SomeClass)`` raised an error already, the developer may sooner realize the cause of the problem.

To summarize, here is an example:
```python
jsons.load(None, str)
```
Current output:
```python
None
```
Desired output:
``DeserializationError`` is raised.

This was also mentioned in [Issue#56](https://github.com/ramonhagenaars/jsons/issues/56).

**Comment by ramonhagenaars** [comment]
Implemented in [v0.9.0](https://pypi.org/project/jsons/).
